### ISSUE TITLE
Joystick isn't correctly managed on CPC

### ISSUE THREAD

**Issue by ldir-hector** [spotlessmind1975/ugbasic#527]
When I run this program :
```
DO
    IF JLEFT(1) THEN 
    				PRINT "WEST"
    			ENDIF	
    IF JRIGHT(1) THEN 
    				PRINT "EAST"
    			ENDIF	
    IF JUP(1) THEN 
    			PRINT "NORTH"
    		ENDIF	
    IF JDOWN(1) THEN 
    				PRINT "SOUTH"
    			ENDIF
    IF JFIRE(1) THEN 
    				CENTRE "BANG!"
    			ENDIF
 LOOP
```
I got the following result showing that everything is pressed at the same time. 
![image](https://user-images.githubusercontent.com/118670680/204111675-0d8726a6-5d5f-43d0-9f85-95e95cbac461.png)

If i reset the emulator to go back to the standard basic I can see that the joystick is correctly detected : 
![image](https://user-images.githubusercontent.com/118670680/204111685-ba19b0a2-972f-42f7-ba92-2e960897b967.png)

I got the same kind of behavior with this simple loop 
```
DO
    j=JOY(1)
    PRINT BIN$(j,5),j
 LOOP
```

that doesn't reflect any of my button smashing.

**Comment by spotlessmind1975** [comment]
Hello @ldir-hector, and **for reporting the problem!**

This is actually a bug, and I think related to the hooking of an incorrect joystick management routine. 
It will be corrected as soon as possible.

**Comment by spotlessmind1975** [comment]
The bug has been fixed - **COLDFIX v1.12 [rev. 20221201.0129]**.
